common/src/Polyhedron_Face.h

```diff
@@ -333,96 +331,64 @@ void Polyhedron<T,FP,VP>::Face::flip() {
+    countAndSetFace(edges.front(), edges.back(), this);
+    m_boundary.insert(HalfEdgeList::iter(after->next()), std::forward<H>(edges));
 }
 
 template <typename T, typename FP, typename VP>
-size_t Polyhedron<T,FP,VP>::Face::removeFromBoundary(HalfEdge* from, HalfEdge* to) {
+typename Polyhedron<T,FP,VP>::HalfEdgeList Polyhedron<T,FP,VP>::Face::removeFromBoundary(HalfEdge* from, HalfEdge* to) {
     ensure(from != nullptr, "from is null");
     ensure(to != nullptr, "to is null");
     assert(from->face() == this);
     assert(to->face() == this);
 
-    const auto removeCount = countAndUnsetFace(from, to->next());
-    m_boundary.remove(from, to, removeCount);
-    return removeCount;
+    const auto removeCount = countAndUnsetFace(from, to);
+    return m_boundary.remove(HalfEdgeList::iter(from), std::next(HalfEdgeList::iter(to)), removeCount);
 }
 
 template <typename T, typename FP, typename VP>
-size_t Polyhedron<T,FP,VP>::Face::removeFromBoundary(HalfEdge* edge) {
-    removeFromBoundary(edge, edge);
-    return 1;
+typename Polyhedron<T,FP,VP>::HalfEdgeList Polyhedron<T,FP,VP>::Face::removeFromBoundary(HalfEdge* edge) {
+    return removeFromBoundary(edge, edge);
 }
 
-template <typename T, typename FP, typename VP>
-size_t Polyhedron<T,FP,VP>::Face::replaceBoundary(HalfEdge* edge, HalfEdge* with) {
-    return replaceBoundary(edge, edge, with);
-}
-
-template <typename T, typename FP, typename VP>
-size_t Polyhedron<T,FP,VP>::Face::replaceBoundary(HalfEdge* from, HalfEdge* to, HalfEdge* with) {
+template <typename T, typename FP, typename VP> template <typename H>
+typename Polyhedron<T,FP,VP>::HalfEdgeList Polyhedron<T,FP,VP>::Face::replaceBoundary(HalfEdge* from, HalfEdge* to, H&& with) {
     ensure(from != nullptr, "from is null");
     ensure(to != nullptr, "to is null");
-    ensure(with != nullptr, "with is null");
     assert(from->face() == this);
     assert(to->face() == this);
-    assert(with->face() == nullptr);
-
-    const auto removeCount = countAndUnsetFace(from, to->next());
-    const auto insertCount = countAndSetFace(with, with, this);
-    m_boundary.replace(from, to, removeCount, with, insertCount);
-    return removeCount;
-}
 
-template <typename T, typename FP, typename VP>
-void Polyhedron<T,FP,VP>::Face::replaceEntireBoundary(HalfEdgeList& newBoundary) {
-    using std::swap;
-
-    unsetBoundaryFaces();
-    swap(m_boundary, newBoundary);
-    setBoundaryFaces();
+    const auto removeCount = countAndUnsetFace(from, to);
+    countAndSetFace(with.front(), with.back(), this);
+    return m_boundary.splice_replace(HalfEdgeList::iter(from), std::next(HalfEdgeList::iter(to)), removeCount, std::forward<H>(with));
 }
 
 template <typename T, typename FP, typename VP>
-size_t Polyhedron<T,FP,VP>::Face::countAndSetFace(HalfEdge* from, HalfEdge* until, Face* face) {
-    size_t count = 0;
+size_t Polyhedron<T,FP,VP>::Face::countAndSetFace(HalfEdge* from, HalfEdge* to, Face* face) {
```

Comment:
These were changed from exclusive of `until` to inclusive to `to`?

Reply:
Yes, although the wording didn't make this clear. But the functions taking ranges of half edges in `Polyhedron::Face` all take the end of the range inclusively. It is debatable if we should change this to take iterators instead, but that would incur more changes at the call sites, where we usually deal with inclusive ranges.